packages/other/src/DragList.ts

```diff
@@ -0,0 +1,200 @@
+import { HTMLWidget, mouse as d3Mouse, select as d3Select, selectAll as d3SelectAll, Widget } from "@hpcc-js/common";
+
+export class DragList extends HTMLWidget {
+
+    protected _container;
+    protected _containerTop;
+    protected _dragMemory: any = {};
+
+    constructor() {
+        super();
+    }
+
+    removeItem(node: HTMLElement) {
+        const item = node.parentElement.parentElement;
+        const order = Number(item.getAttribute("data-order"));
+        this.widgets().splice(order, 1);
+        this.lazyRender();
+    }
+
+    applySortOrder() {
+        const order = d3SelectAll(".drag-list-item").nodes()
+            .map((elm: HTMLElement) => Number(elm.getAttribute("data-order")))
+            ;
+
+        const orderedWidgetArr = [...this.widgets()];
+
+        order.forEach((val, i) => { orderedWidgetArr[val] = this.widgets()[i]; });
+
+        this.widgets(orderedWidgetArr);
+        this.lazyRender();
+    }
```

Comment:
There is an easier way to do this. If the Array of "data" that is being used to generate the list is in the correct order, then you can get d3 to ensure the web items match by calling selection.order().

Reply:
This method reorders the data to match the order of the dom elements. Based on the d3 documentation it seems like `selection.order()` is for doing the opposite.

Reply:
Ahh ok!